src/Analyzer/Passes/RewriteAggregateFunctionWithIfPass.cpp

```diff
@@ -74,7 +74,13 @@ class RewriteAggregateFunctionWithIfVisitor : public InDepthQueryTreeVisitorWith
 
                 /// We need to preserve the output type from if()
                 if (if_arguments_nodes[1]->getResultType()->getName() != if_node->getResultType()->getName())
+                {
+                    /// For count, adding an extra cast is too heavy for the benefit.
```

Comment:
Can't we just not add the cast? The return type of count() is the same either way, right?

Reply:
Good idea! It doesn't change return type of count() whether cast is added or not.